src/structures/guild_preference.ts

```diff
@@ -671,26 +709,40 @@ export default class GuildPreference {
 
     /** Resets the multiguess type option the the default value */
     async resetMultiGuessType() {
-        this.gameOptions.multiGuessType = DEFAULT_MULTIGUESS_TYPE;
-        await this.updateGuildPreferences(true);
+        await this.setMultiGuessType(DEFAULT_MULTIGUESS_TYPE);
     }
 
     /**
      * Persists the current guild preference to the data store
+     * @param updatedOptionsObjects - An array of objects containing the names and values of updated options
      */
-    async updateGuildPreferences(updateGameSession: boolean) {
-        await dbContext.kmq("guild_preferences")
-            .where({ guild_id: this.guildID })
-            .update({ guild_preference: JSON.stringify(this) });
+    async updateGuildPreferences(updatedOptionsObjects: Array<{ name: string, value: GameOptionValue }>) {
+        const updatedOptions = Object.values(updatedOptionsObjects).map((option) => ({
+            guild_id: this.guildID,
+            option_name: option.name,
+            option_value: JSON.stringify(option.value),
+        }));
+        await dbContext.kmq.transaction(async (trx) => {
+            await dbContext.kmq("game_options")
+                .insert(updatedOptions)
+                .onConflict(["guild_id", "option_name"])
+                .merge()
+                .transacting(trx);
+        });
         const gameSession = state.gameSessions[this.guildID];
-        if (gameSession && updateGameSession) {
+        if (gameSession) {
```

Comment:
Why was the `updateGameSession` removed here?

Reply:
Previously we called this function with `updateGameSession = false` in only one place, `fromGuild`. However, since fromGuild is passed with whatever is in the database already, it doesn't need to call `updateGuildPreferences`. For every remaining call of `updateGuildPreferences`, an active game session would need to be updated